services/user/Supervisor/ui/src/oauth/db.ts

```diff
@@ -0,0 +1,59 @@
+import { Result } from '../hostInterface';
+import { RecoverableErrorPayload } from '../plugin/errors';
+import { OAUTH_REQUEST_KEY, OAUTH_REQUEST_EXPIRATION } from '../constants';
+import { getSupervisor } from "@psibase/common-lib";
+import { z } from 'zod';
+
+export const supervisor = getSupervisor();
+
+export const OauthRequestSchema = z.object({
+    id: z.string().uuid(),
+    subdomain: z.string(),
+    subpath: z.string(),
+    expiry_timestamp: z.number(),
+});
+
+export type OauthRequest = z.infer<typeof OauthRequestSchema>;
+
+export class ActiveOauthRequest {
+    static async get(id: string): Promise<Result<OauthRequest, RecoverableErrorPayload>> {
+        const oauthReqBytes = await supervisor.functionCall({
+            service: "clientdata",
+            intf: "keyvalue",
+            method: "get",
+            params: [OAUTH_REQUEST_KEY]});
+
+        if (!oauthReqBytes) {
+            throw new Error("No active oauth request found");
+        }
+
+        const oauthReq = JSON.parse(new TextDecoder().decode(oauthReqBytes));
+        if (id != oauthReq.id) {
+            await this.delete();
+            throw new Error("Oauth request id mismatch");
+        }
+
+        let isExpired = (Math.floor(new Date().getTime() / 1000)
+            - oauthReq.expiry_timestamp.seconds)
+            >= OAUTH_REQUEST_EXPIRATION;
+        if (isExpired) {
+            await this.delete();
+            throw new Error("Oauth request expired");
+        }
```

Comment:
This seems strange as the expiry_timestamp is more, `created_at`.
I think it would be more common to treat this more as a traditional expiry timestamp and then add the `OAUTH_REQUEST_EXPIRATION` when this is made rather than assessed. 

```suggestion
        let isExpired = (Math.floor(new Date().getTime() / 1000)
            >= oauthReq.expiry_timestamp.seconds);
           if (isExpired) {
            await this.delete();
            throw new Error("Oauth request expired");
        }
```